cmd/regex_format_test.go

```diff
@@ -513,6 +513,35 @@ this is a regex
 	s.Contains(out.String(), "{1,3}[Bb]lah\\n======^ [HERE]\\n\"}\n")
 }
 
+func (s *formatTestSuite) TestFormat_IgnoreCaseFlagWithUppercase() {
+	// send logs to buffer
+	out := &bytes.Buffer{}
+	log := zerolog.New(out)
+	logger = log.With().Str("component", "parser-test").Logger()
+
+	s.writeDataFile("123456.ra", regexAssemblyStandardHeader+`
+##!+ i
+this is a regex
+^[a-z]this is another regex
+{1,3}[Bb]lah
+`)
+	rootCmd.SetArgs([]string{"-d", s.tempDir, "regex", "format", "123456"})
+
+	_, err := rootCmd.ExecuteC()
+	s.NoError(err)
+	s.Contains(out.String(), "File contains uppercase letters, but ignore-case flag is set. Please check your source files.")
+	s.Contains(out.String(), "{1,3}[Bb]lah\\n======^ [HERE]\\n\"}\n")
+
+	expected := regexAssemblyStandardHeader + `
+##!+ i
+this is a regex
+^[a-z]this is another regex
+{1,3}[bb]lah
```

Comment:
Ugh, I get we are lowercasing, but this might be problematic. Why would we want something like `[bb]`?

Reply:
I consciously ignored that problem :D. I agree, it would be nice to fix those character classes and I think it wouldn't be too hard to do it. The reason I didn't tackle it is that those kinds of character classes are still semantically correct and would be reduced by the assembly step. I'll add that functionality.